.github/workflows/auto-update.yml

```diff
@@ -50,21 +50,24 @@ jobs:
           cd graypaper
           echo "VERSION=v$(git describe --tags --abbrev=0)" >> "$GITHUB_OUTPUT"
           cd ..
-      - name: Commit changes to the main branch
-        uses: EndBug/add-and-commit@v9
-        with:
-          default_author: github_actions
-          message: Bump to ${{steps.graypaper_update.outputs.VERSION}}
-          tag: ${{steps.graypaper_update.outputs.VERSION}}
+      - name: Use git as github actions bot
+        run: |
+          BOT_NAME="github-actions[bot]"
+          BOT_EMAIL="$(gh api /users/github-actions[bot] | jq -r '.id')+$BOT_NAME@users.noreply.github.com"
+          git config --global user.name $BOT_NAME
+          git config --global user.email $BOT_EMAIL
+      - name: Create a PR and merge changes to the main branch
+        run: |
+          git checkout -b feature/${{steps.graypaper_update.outputs.VERSION}}
+          git add .
+          git commit -m "Bumped Graypaper to ${{steps.graypaper_update.outputs.VERSION}}"
+          git push --set-upstream origin feature/${{steps.graypaper_update.outputs.VERSION}}
+          gh pr create -B main -H feature/$VERSION \
+            --title "Bump Graypaper to ${{steps.graypaper_update.outputs.VERSION}}" --body ""
+          gh pr merge --merge --delete-branch
       - name: Create Release
-        id: create_release
-        uses: actions/create-release@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          tag_name: ${{ steps.graypaper_update.outputs.VERSION }}
-          release_name: Release ${{ steps.graypaper_update.outputs.VERSION }}
-          body: Bumped Graypaper version to ${{steps.graypaper_update.outputs.VERSION}}
-          draft: false
-          prerelease: false
+        run: |
+          gh release create ${{ steps.graypaper_update.outputs.VERSION }} \
```

Comment:
Shouldn't we wait with the release till the merge is done? What code will be referenced in the release? Is it going to be the PR content or what was merged to the main branch?

Reply:
afaik `gh pr merge` is synchronous therefore the release should only happen when merge was successful.

As for what code will be referenced this is a quote from `gh release` docs:

> If a matching git tag does not yet exist, one will automatically get created from the latest state of the default branch